cmd/gateway/main.go

```diff
@@ -76,24 +88,32 @@ func main() {
 
 	apiMux.Handle("/metrics", promhttp.Handler())
 
-	go func() {
-		logger.Log.Info("Starting management web server:  ", *mgmtAddr)
-		if err := http.ListenAndServe(*mgmtAddr, handlers.LoggingHandler(os.Stdout, apiMux)); err != nil {
-			logger.Log.WithFields(logrus.Fields{"error": err}).Fatal("managment web server error")
-		}
-	}()
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
 
-	go func() {
-		logger.Log.Info("Starting websocket server on:  ", *wsAddr)
-		if err := http.ListenAndServe(*wsAddr, handlers.LoggingHandler(os.Stdout, wsMux)); err != nil {
-			logger.Log.WithFields(logrus.Fields{"error": err}).Fatal("websocket server error")
-		}
-	}()
+	apiSrv := utils.StartHTTPServer(*mgmtAddr, "management", apiMux)
+	wsSrv := utils.StartHTTPServer(*wsAddr, "websocket", wsMux)
+	wsSrv.RegisterOnShutdown(func() { closeConnections(cm, wg, cfg.HttpShutdownTimeout) })
 
 	signalChan := make(chan os.Signal, 1)
+	done := make(chan bool, 1)
 
 	signal.Notify(signalChan, syscall.SIGINT, syscall.SIGTERM)
 
-	<-signalChan
-	logger.Log.Debug("Receptor-Controller shutting down")
+	go func() {
```

Comment:
I ran some tests where I removed this go routine and the done channel and just had the main thread wait for the signalChan and waitgroup.  It seemed to work fine with that approach.  To me, that seems a bit simpler to me to wrap my head around.  What do you think?  Does that approach make sense?  Is there a downside to that approach?

Reply:
I was using the done channel with the go routine before I added the wait group. Had I done the wait group first I probably never would've added it. Both approaches should work totally fine, and I agree that ditching the extra channel and go routine would be a bit simpler. I'll update!